lxd/device/device_instance_id.go

```diff
@@ -15,4 +16,5 @@ type InstanceIdentifier interface {
 	LogPath() string
 	ExpandedConfig() map[string]string
 	ExpandedDevices() config.Devices
+	CurrentIdmap() (*idmap.IdmapSet, error)
```

Comment:
I suspect we'll need to abstract that further later as that's a rather container-specific concept.

Reply:
@stgraber no problem, I've removed that from the interface and added shifting support to the `c.deviceStart` function by adding a new `c.deviceShiftMounts` function, which will then do the same thing as what `device.UnixCreateDevice` does when passed a non-nil idmap set.